noticeboardapp/views.py

```diff
@@ -0,0 +1,158 @@
+from django.shortcuts import render, redirect
+from datetime import date, datetime
+from noticeboardapp.models import Notice
+from django.contrib.auth.decorators import login_required
+
+
+
+# Create your views here.
+
+
+@login_required
+def add_notice_view(request):
```

Comment:
Look at create view
https://akademiait.com.pl/class-based-views-w-django-wygodny-sposob-tworzenia-widokow-czesc-2/
Will simplify it a lot. 
And of course move to forms.